### ISSUE TITLE
Write the GUID in the PSet file in the IFC notation

### ISSUE THREAD

**Issue by klacol** [BuildingSMART/IfcDoc#30]
The IFC Documentation includes a list of files, that can be downloaded for serveral purposes. This can be done here:

http://www.buildingsmart-tech.org/ifc/IFC4/Add2/html/
then A. Computer interpretable listings
then A.1 General Usage
then Property and quantity templates

The templates can be downloaded in two formats:

IFC-SPF property and quantity templates	
http://www.buildingsmart-tech.org/ifc/IFC4/Add2/html/annex/annex-a/general-usage/IFC4_ADD2.ifc

PSD-XML property templates in ZIP file
http://www.buildingsmart-tech.org/ifc/IFC4/Add2/html/annex/annex-a/general-usage/IFC4_ADD2-psd.psdzip

In both files the templates contain the GUID of the IfcPropertyTemplate and they correlate to the GUID's in bSDD.

Example
Pset: Pset_BeamCommon
Property: Reference

Guid in the Pset-File: ifdguid="04abf900d1c411e1800000215ad4efdf"
Guid in the IFC file: "04g$a0qSGHuO00025QrE$V"

The Guid in the Pset-File seems to be identical to the GUID in the IFC file, but it is shown in a different notation.

Guid in the IFC file:  "04g$a0qSGHuO00025QrE$V"
Guid in the PSet file: "04abf900d1c411e1800000215ad4efdf"

I have transformed the GUID in the IFC file into a valid GUID in the format "8-4-4-4-12": 
"04abf900-d1c4-11e1-8000-00215ad4efdf"

It seems to be, that the Guid in the PSet file is identical to the transformed IFC-Guid, but without the hyphens.

I propose, tho write the GUID's in both file in the identical IFC notation.

The GUID in the PSet file should be also this one: "04g$a0qSGHuO00025QrE$V"

**Comment by timchipman** [comment]
The XSD only defines the guid as a string, so could really support any encoding (or perhaps multiple encodings), and any RFC variant (as described at https://en.wikipedia.org/wiki/Universally_unique_identifier). Unfortunately, documentation was never published for the PSD XML format -- not sure if it was ever meant to be used outside of internal developments, though it has in practice. Question is - if we change it to use IfcGloballyUniqueId encoding, would that break existing implementations? Or there's the flip side -- should bsDD (or even IfcGloballyUniqueId) change to format GUIDs as hexadecimal per standards and as is done pretty much everywhere else?

The current format is what is typically found in XML encodings -- the IfcGloballyUniqueId encoding is a one-off conversion that was introduced in the days IFC files were stored on 3.5" floppy disks where every byte counted (and before other approaches such as Base64 or zip were commonplace), and remains for compatibility purposes. I don't really know why this formatting convention propagated to other implementations.

**Comment by klacol** [comment]
Yes, that are the possibilities. Since the transformation of an "IfcGuid" to a "StandardGuid" can only be done with a tool. It is complicated to transform it with SQL or Excel, since the formula for the transformation is quite complicated.

The question, if we change to a "StandardGuid" is a big question and can only be answered in coordination with the bSDD and others. Until then, the format in the Pset-File should be the IfcGuid IMHO.
